app/models/from_address.rb

```diff
@@ -33,6 +36,13 @@ def default_email?
     email_address == DEFAULT_EMAIL_FROM
   end
 
+  def allowed_domain?(email)
```

Comment:
sorry should have brought this up yesterday. the `from_address` object already has the email as an accessor therefore it doesn't need to be passed into this method, especially as the `from_address_creation` has access to the specific `from_address` object containing the email.

Therefore this can be a method that acts on the attribute inside the `from_address` object and then you can just delegate the `allowed_domain?` method to the `from_address` object in the `from_address_creation` model.